package-res/resources/web/pentaho/data/filter/AbstractFilter.js

```diff
@@ -0,0 +1,189 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../../lang/Base",
+  "../../util/arg",
+  "require"
+], function(Base, arg, require,  Or, And, Not) {
+  "use strict";
+
+  /**
+   * @name AbstractFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @abstract
+   * @amd pentaho/data/filter/AbstractFilter
+   *
+   * @classdesc This tree structure acts as a filter for a data set and is made up of:
+   *
+   * * Non-leaf nodes
+   * * Leaf nodes
+   *
+   * Each non-leaf node can be seen as a filter that acts upon its leaf nodes
+   * (the atomic predicates of the filtering) and aggregates the outcomes of each. In this
+   * way the structure is able to provide an intensional representation of a selection of
+   * the given data set.
+   *
+   * @description The `AbstractFilter` class is the abstract base class of
+   * classes that represent a data set filter.
+   *
+   * ### Remarks
+   *
+   * The following derived classes are also abstract and provide the non-leaf and leaf
+   * nodes respectively that override and implement the methods necessary for filtering a
+   * data table {@link pentaho.data.Table}.
+   *
+   * * {@link pentaho.data.filter.AbstractPropertyFilter}
+   * * {@link pentaho.data.filter.AbstractTreeFilter}
+   *
+   * NOTE: Other derived classes could be implemented in order to support filtering of other data formats.
```

Comment:
I would leave this comment out for now, at least until we ascertain if we are going to support data formats other than `pentaho.data.Table`